xmppserver/src/main/java/org/jivesoftware/openfire/session/LocalSession.java

```diff
@@ -135,16 +142,26 @@ public LocalSession(String serverName, Connection connection, StreamID streamID,
      * @return true if session detached
      */
     public boolean isDetached() {
-        return this.conn == null;
+        lock.readLock().lock();
```

Comment:
This lock achieves nothing. The conn is either null or not at the time of checking, so no inconsistency or other issues here.

Reply:
done